cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -767,3 +765,111 @@ func convertToStackParams(parameters azure.ArmParameters) map[string]*armdeploym
 	}
 	return stackParams
 }
+
+func (d *StackDeployments) ValidatePreflightToResourceGroup(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	client, err := d.createClient(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	stack, err := d.stackFromArmForResourceGroup(ctx, subscriptionId, armTemplate, parameters, tags, options)
+	if err != nil {
+		return err
+	}
+
+	var rawResponse *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
```

Comment:
I'm seeing that this method is deprecated, does `policy.WithCaptureResponse` work here instead? I see we use it in `standard_deployments.go`.

> WithCaptureResponse applies the HTTP response retrieval annotation to the parent context. The resp parameter will contain the HTTP response after the request has completed. Deprecated: use [policy.WithCaptureResponse] instead.